### PR TITLE
Issue #837: Add §23 non-contiguous git invocation heuristic to verify-patterns.md

### LINKED ISSUE #837: issue: verify command 生成時の git invocation contiguous sub-string heuristic を追加
## Background

`/issue` skill が verify command を生成する際、`file_contains` / `grep` で git invocation の特定文字列を検証するケースで、コードが `git -C "$REPO_ROOT" commit` のように非連続な形式で書かれていると `file_contains "scripts/foo.sh" "git commit"` がマッチせず、verify command が FAIL になる問題が頻発している。

本セッションで観察された連続事例:
- **#823**: `tests/auto-completion-report.bats` の verify command を初回設定で間違え、code phase で修正コミット 1 件追加。
- **#824**: AC1 verify command `file_contains "scripts/append-loop-state-heartbeat.sh" "git commit"` が `git -C "$REPO_ROOT" commit` と非連続なためマッチせず、code phase で `"commit -s"` に修正する追加コミットが必要だった。

検出元 (#824 verify retrospective):
> `git -C "$REPO_ROOT" commit` 等の git invocation 形式が `file_contains "git commit"` verify command にマッチしない問題が code phase で 1 commit 分の修正コストを発生させた。

## Purpose

`/issue` skill (または verify command 生成を担う module) に "git invocation や類似の non-contiguous シンボル参照は contiguous な sub-string (`commit -s`, `-m "Add ..."` 等) を選ぶ" 旨のガイドラインを追加し、verify command の calibration ミスによる code phase 修正コストを削減する。

## Proposal (Outline)

**A. `/issue` SKILL.md に verify command 生成 heuristic 追記**
- "git invocation のシンボル (`git -C "$REPO_ROOT" commit` 等) を `file_contains` / `grep` でチェックする場合、`git commit` のような contiguous 形式ではマッチしないことがある。`commit -s` / `-m "Add ..."` 等の確実に連続する sub-string を選ぶこと" のガイドラインを heuristic section に追加。

**B. `modules/verify-patterns.md` または `modules/verify-classifier.md` / `modules/ambiguity-detector.md` に同 heuristic を集約**
- `/issue` だけでなく `/spec` でも verify command を生成・修正する場面があるため module 化が望ましい。
- `modules/verify-patterns.md` は verify command accuracy guidelines の SSoT であるため最有力候補。

**C. `tests/` 配下に regression test 追加**
- 上記 heuristic が SKILL.md / module に存在することを structural assertion で保証。
- 新規作成: `tests/verify-heuristics.bats`

### Pre-merge (auto-verified)

- [x] verify command 生成時の git invocation heuristic (non-contiguous シンボルを `file_contains` / `grep` でチェックする際に contiguous sub-string を選ぶ旨のガイドライン) が文書化されている 
- [x] `tests/verify-heuristics.bats` (新規作成) が追加され、heuristic guideline の structural regression test が通過する 

### Post-merge

- [x] 次回 `/auto` (または `/issue`) 実行時、git invocation を含む実装の verify command が contiguous sub-string で生成されることを観察 

## Auto-Resolved Ambiguity Points

- **対象ファイル候補**: 元の rubric は `skills/issue/SKILL.md`、`modules/verify-classifier.md`、`modules/ambiguity-detector.md` の 3 ファイルのみを対象としていたが、`modules/verify-patterns.md` が verify command accuracy guidelines の SSoT として最適候補であるため rubric 対象に追加した。
- **regression test (Proposal C)**: 提案に含まれるが元の AC に不記載であった。`tests/verify-heuristics.bats` (新規作成) を Pre-merge AC として追加した。

## Related

- 発生元: #824 verify retrospective (再発性: #823 でも同種 calibration 問題発生)
- 関連 Skill: `skills/issue/SKILL.md`
- 関連 module: `modules/verify-patterns.md`, `modules/verify-classifier.md`, `modules/ambiguity-detector.md`
- 関連 Issue: #827 (behavioral changes 時の verify command broader scope) と相補関係

### PR DESCRIPTION
**Pull request by saito**
## Summary

- `modules/verify-patterns.md` に §23「Non-Contiguous Git Invocation — Prefer Contiguous Sub-strings」を追加
- `file_contains` / `grep` で `git -C "$REPO_ROOT" commit` 等の非連続 git invocation を検証する際に `git commit` がマッチしない問題に対するガイドラインを文書化
- `tests/verify-heuristics.bats` を新規作成し、§23 ガイドラインの structural regression test を追加 (3 テスト)
- `docs/structure.md` および `docs/ja/structure.md` の tests/ ファイル数を 91 → 93 に更新

## Verification (pre-merge)

- modules/verify-patterns.md に §23 が追加され、non-contiguous シンボルの contiguous sub-string 選択ガイドラインが文書化されている
- `bats tests/verify-heuristics.bats` が 3/3 PASS (structural regression test)

## Verification (post-merge)

- 次回 `/auto` または `/issue` 実行時、git invocation を含む実装の verify command が contiguous sub-string で生成されることを観察

closes #837

### CODE REVIEW

File: modules/verify-patterns.md [hunk 1]
```diff
@@ -801,6 +801,45 @@ The first condition verifies installation (mechanical, dependency-free). The sec
...
+
+**Generalizes to any command with inserted flags:**
+
+This pattern applies whenever option flags are inserted between a command name and its sub-command:
+
+| Full form (non-contiguous) | Contiguous anchor |
+|---------------------------|-------------------|
+| `git -C "$REPO_ROOT" commit` | `commit -s` / `commit -m` |
+| `git -C "$REPO_ROOT" push` | `push origin` |
+| `ssh -i key host command` | `host command` (the host+command part) |
```
Comment (saito) [inline modules/verify-patterns.md]: **[Spec Deviation] modules/verify-patterns.md:835** — `ssh` 例の contiguous anchor がプレースホルダー文字列

`ssh -i key host command` 行で contiguous anchor として `host command` と記載されているが、`host` と `command` はプレースホルダーであり、実際の ssh コマンドではこのリテラル文字列は存在しない。ガイドラインを理解する助けにならず、むしろ混乱を招く可能性がある。Issue Severity: SHOULD

**Recommended fix:**
この行を削除するか、より具体的な実例 (e.g., `ssh -i key myserver.example.com "deploy.sh"` → anchor: `"deploy.sh"`) に差し替える。または表の Note 列に「`host`/`command` を実際のリテラル文字列に置き換えること」という注記を追加する。

File: modules/verify-patterns.md [hunk 2]
```diff
@@ -801,6 +801,45 @@ The first condition verifies installation (mechanical, dependency-free). The sec
...
+
+**Decision procedure:**
+
+1. Before writing `file_contains "path" "git commit"`, check whether the implementation may use `git -C` or other inserted flags
```
Comment (saito) [inline modules/verify-patterns.md]: **[Spec Deviation] modules/verify-patterns.md:839** — Decision Procedure のステップ 1 が `git commit` に限定されており、セクション本文の「汎化」と矛盾

ステップ 1 の記述は `file_contains "path" "git commit"` に限定されているが、§23 本文では「any command with inserted flags に汎化される」と明記し、`git push` / `ssh -i` の例も示している。読者がステップ 1 を読むと `git push` や `ssh` では本ヒューリスティックを適用しなくてよいと誤解するリスクがある。Issue Severity: SHOULD

**Recommended fix:**
ステップ 1 を汎化する:
```
1. Before writing `file_contains "path" "git commit"` (or any pattern matching `<command> <subcommand>` where option flags may be inserted between them), check whether the implementation may use `git -C`, `ssh -i`, or other flag-insertion forms
```

File: modules/verify-patterns.md [hunk 3]
```diff
@@ -801,6 +801,45 @@ The first condition verifies installation (mechanical, dependency-free). The sec
...
+This FAILs when the actual code is `git -C "$REPO_ROOT" commit -s -m "..."` — `git commit` is split by the `-C "$REPO_ROOT"` flag and does not appear on any single line.
+
+**Recommended patterns:**
+
+```
+file_contains "scripts/foo.sh" "commit -s"
+file_contains "scripts/foo.sh" "commit -m"
+```
+
+Both `commit -s` and `commit -m` are contiguous sub-strings regardless of what precedes `commit` on the line.
```
Comment (saito) [inline modules/verify-patterns.md]: **[Edge Case/Robustness] modules/verify-patterns.md:823** — `commit -s` anchor が signed-off commit を使わないプロジェクトでは不在

`commit -s` と `commit -m` を並列に「recommended patterns」として挙げているが、`commit -s` は DCO/signed-off が有効なプロジェクト固有の慣習。wholework 以外のプロジェクトでは `-s` フラグを使わないスクリプトが多く、ガイドライン読者が wholework の実例に倣って `commit -s` を採用すると verify command が FAIL する。Issue Severity: SHOULD

**Recommended fix:**
この説明に「プロジェクトで signed-off commit を使っている場合は `commit -s`、使っていない場合は `commit -m` を選ぶ — anchor は実装ファイルの実態に合わせること」という一文を追加して、選択がプロジェクト依存であることを明示する。

File: tests/verify-heuristics.bats [hunk 4]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bats
+# Structural regression tests for verify command heuristic guidelines.
+PROJECT_ROOT="$(cd "$(dirname "$BATS_TEST_FILENAME")/.." && pwd)"
+VERIFY_PATTERNS="$PROJECT_ROOT/modules/verify-patterns.md"
+
+@test "verify-heuristics: non-contiguous heuristic section exists in verify-patterns.md" {
+    grep -q "non-contiguous" "$VERIFY_PATTERNS"
```
Comment (saito) [inline tests/verify-heuristics.bats]: **[Edge Case/Robustness] tests/verify-heuristics.bats:7** — テスト対象 keyword の specificity が低く将来的な偽陽性リスク

`grep -q "non-contiguous"` 等の keyword 存在チェックは、§23 が削除されても別セクションに同じ keyword が存在すれば PASS する。現時点では `non-contiguous` と `git -C` は §23 以外に存在しないため実質リスクはゼロだが、将来キーワードが別文脈で追加されると regression が素通りする。Issue Severity: CONSIDER

**Recommended fix:**
セクション見出しを直接検証する anchor に変更する:
```bash
grep -q "Non-Contiguous Git Invocation" "$VERIFY_PATTERNS"
```
セクション見出し自体を anchor にすることで、§23 の削除に対して耐性が高くなる。

**Review by saito — COMMENTED** [review]
## Acceptance Criteria Verification Results

| Condition | Result | Notes |
|-----------|--------|-------|
| git invocation heuristic (contiguous sub-string selection guideline) が文書化されている | PASS | modules/verify-patterns.md に §23 追加済み |
| `tests/verify-heuristics.bats` が追加され structural regression test が通過 | PASS | CI `Run bats tests` SUCCESS (CI reference fallback) |
| 次回 /auto 実行時に contiguous sub-string で verify command 生成されることを観察 | POST-MERGE | `verify-type: observation event=auto-run` — merge 後に観察 |

## CI Status

| Job name | Status | Notes |
|---------|--------|-------|
| DCO | SUCCESS | — |
| Run bats tests | SUCCESS | — |
| Validate skill syntax | SUCCESS | — |
| Forbidden Expressions check | SUCCESS | — |
| macOS shell compatibility | SUCCESS | — |

## Code Review

### General Comments

特定ファイル/行に帰属できない設計レベルの指摘はありません。

### Issue Count Summary

- Posted as line comments: 4 (MUST: 0, SHOULD: 3, CONSIDER: 1)
- General Comments (in body): 0

---
Next: run `/merge 840`.

**Comment by saito** [comment]
Review feedback addressed in this commit:
1. SHOULD: `commit -s` anchor clarification — added note that `commit -s` requires DCO/signed-off projects, use `commit -m` otherwise
2. SHOULD: ssh placeholder example — replaced `host command` placeholder with concrete example `"deploy.sh"`
3. SHOULD: Decision Procedure step 1 generalized to cover any `<command> <subcommand>` pattern (not just `git commit`)
4. CONSIDER: `verify-heuristics.bats` first test now uses section heading `Non-Contiguous Git Invocation` as anchor (higher specificity)

**Comment by saito** [comment]
## Review Response Summary

### Claude Review Response

| Aspect | Filename | Line | Issue | Response | Fix/Skip reason |
|--------|---------|------|-------|----------|----------------|
| Spec Deviation | modules/verify-patterns.md | 839 | Decision Procedure step 1 が `git commit` のみ対象で本文の「汎化」と矛盾 | Resolved | `<command> <subcommand>` 全般に適用可能な記述に変更 |
| Spec Deviation | modules/verify-patterns.md | 835 | ssh 例の contiguous anchor がプレースホルダー文字列でミスリーディング | Resolved | 具体的な `"deploy.sh"` anchor 例に差し替え |
| Edge Case/Robustness | modules/verify-patterns.md | 823 | `commit -s` が signed-off commit を使わないプロジェクトで誤用される可能性 | Resolved | DCO プロジェクトは `commit -s`、それ以外は `commit -m` と使い分けを明示 |
| Edge Case/Robustness | tests/verify-heuristics.bats | 7 | keyword の specificity が低く将来的な偽陽性リスク | Resolved | section heading `Non-Contiguous Git Invocation` を anchor に変更 |

### Lightweight Re-check Results

- `validate-skill-syntax.py`: PASS (0 errors, 1 pre-existing warning — unrelated to this PR)
- Fix diff check: PASS (4 changes in `modules/verify-patterns.md` + 1 change in `tests/verify-heuristics.bats`)
- New issues: none
